File: src/test/java/org/dedda/bratwurst/parse/PushParserTest.java
package org.dedda.bratwurst.parse;

import org.dedda.bratwurst.BratwurtstTestcase;
import org.dedda.bratwurst.lang.BWInteger;
import org.dedda.bratwurst.lang.Push;
import org.testng.annotations.Test;

import static org.testng.Assert.assertEquals;

/**
 * Created by dedda on 12/8/15.
 *
 * @author dedda
 */
public class PushParserTest extends BratwurtstTestcase {

    @Test
    public void testParse() throws Exception {
        int number = 123;
        String line = ">" + number + ">";
        int lineNumber = 12;
        PushParser parser = new PushParser();
        Push push = parser.parse(line, lineNumber);
        assertEquals(new BWInteger(number), push.getArgument());
        assertEquals(lineNumber, push.getLineNumber());
    }
}

File: src/test/java/org/dedda/bratwurst/parse/PushParserTest.kt
package org.dedda.bratwurst.parse

import org.dedda.bratwurst.BratwurtstTestcase
import org.dedda.bratwurst.lang.BWInteger
import org.testng.Assert
import org.testng.annotations.Test

/**
 * Created by dedda on 12/8/15.
 *
 * @author dedda
 */
class PushParserTest : BratwurtstTestcase() {
    @Test
    fun testParse() {
        val number = 123
        val line = ">$number>"
        val lineNumber = 12
        val parser = PushParser()
        val push = parser.parse(line, lineNumber)
        Assert.assertEquals(BWInteger(number), push!!.argument)
        Assert.assertEquals(lineNumber, push.lineNumber)
    }
}